src/main/java/org/opentripplanner/netex/loader/parser/NetexDocumentParser.java

```diff
@@ -0,0 +1,116 @@
+package org.opentripplanner.netex.loader.parser;
+
+import org.opentripplanner.netex.loader.NetexImportDataIndex;
+import org.rutebanken.netex.model.Common_VersionFrameStructure;
+import org.rutebanken.netex.model.CompositeFrame;
+import org.rutebanken.netex.model.GeneralFrame;
+import org.rutebanken.netex.model.InfrastructureFrame;
+import org.rutebanken.netex.model.PublicationDeliveryStructure;
+import org.rutebanken.netex.model.ResourceFrame;
+import org.rutebanken.netex.model.ServiceCalendarFrame;
+import org.rutebanken.netex.model.ServiceFrame;
+import org.rutebanken.netex.model.SiteFrame;
+import org.rutebanken.netex.model.TimetableFrame;
+import org.rutebanken.netex.model.VersionFrameDefaultsStructure;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import javax.xml.bind.JAXBElement;
+import java.util.Collection;
+import java.util.List;
+
+/**
+ * This is the root parser for a Netex XML Document. The parser ONLY read the document and
+ * populate the index with entities. The parser is only responsible for populating the
+ * index, not for validating the document, nor linking of entities or mapping the OTP
+ * internal data structures.
+ */
+public class NetexDocumentParser {
```

Comment:
The `NetexDocumentParser` is not parsing, it would be better to name all the classes and the package `org.opentripplanner.netex.loader.parser` something like `IndexBuilder`, because all thay do is to put the document entities into an index witch is passed on to the mapping code. 

There is also some cases where the _parsers_ do more than that. Especially when accessing other elements from the index the parser relay on the order the elements appear in the xml document - there are no guaranties for such order; Hence we should move such code (mainly building reverse navigation indexes) to a step AFTER the indexing is done.

There are a lot of code not jet submitted as PRs building on the PR, so I would like NOT to change this in this PR - causing a lot of merge conflicts, instead I would like to make a note on this in the #2757.

Reply:
The document would need to be parsed in order to index it, so am I correct in understanding that this indexing code is run after the document is already parsed? That is, it's not a parser, it's run after the parser?

Reply:
Yes, that is safe to say. Iterating trough the XMLDocument generated by the JAXB parser might be close to parsing, but I at least find the names a bit confusing. At the moment parent elements are not considered, so when doing the version interpretation we need to do that, and that would require rethinking some of the design. So it make sence to cleanup as part of the #2781.